jffs2/libjffs2.c

```diff
@@ -185,13 +177,20 @@ static int jffs2_srv_lookup(jffs2_partition_t *p, oid_t *dir, const char *name,
 }
 
 
-static int jffs2_srv_setattr(jffs2_partition_t *p, oid_t *oid, int type, long long attr, void *data, ssize_t size)
+static int libjffs2_setattr(void *info, oid_t *oid, int type, long long attr, void *data, size_t size)
 {
 	struct iattr iattr;
 	struct inode *inode;
 	struct dentry dentry;
 	int ret;
-	struct jffs2_sb_info *c = JFFS2_SB_INFO(p->sb);
+	struct jffs2_sb_info *c;
+	jffs2_partition_t *p;
```

Comment:
You can safely do (it's not a dereference):
```suggestion
	jffs2_partition_t *p = (jffs2_partition_t *)info;
```

Reply:
Done